### ISSUE TITLE
refactor: Use the `E` type for the transformer on Bloc.on<Event>()

### ISSUE THREAD

**Issue by SimoneBressan** [felangel/bloc#2794]
Wouldn't it be more correct that transformer had the type `E`?
```dart
  void on<E extends Event>(
    EventHandler<E, State> handler, {
    EventTransformer<Event>? transformer,
  }) {
```

So it would be:
```dart
  void on<E extends Event>(
    EventHandler<E, State> handler, {
    EventTransformer<E>? transformer,
  }) {
```
and
```dart
final subscription = _transformer(
      _eventController.stream.where((event) => event is E).cast<E>(),
      (dynamic event) {
```

So you can create transformers for specific events. Such as deciding based on an event whether to have sequential or parallel processing or much more

**Comment by felangel** [comment]
Hey @SimoneBressan 👋 
Thanks for opening an issue!

Yup I was just looking at that yesterday as well...will update shortly, thanks! 🙏

**Comment by felangel** [comment]
@SimoneBressan this should be fixed in [bloc v7.2.1](https://pub.dev/packages/bloc/versions/7.2.1).
